src/theme/useThemeToggle.js

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react';
+import { DARK_MODE, LIGHT_MODE, CURRENT_THEME } from '../common/constants';
+
```

Comment:
It may and should be an isolated simple pair of state nd state changer with a grain of locla storage handling
```
export const useThemeChange = (defaultTheme = LIGHT) => {
    const [theme, setTheme] = useState(window.localStorage.getItem(CURRENT_THEME) || defaultTheme);
    
    const changeTheme = theme => {
          window.localStorage.setItem(CURRENT_THEME, theme);
          setTheme(theme);
    };
     
    return [theme, changeTheme];
};
```